packages/juggler/src/loopback-datasource-juggler.ts

```diff
@@ -0,0 +1,662 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { Class, Options, Callback, AnyObject, AnyType } from './common';
+
+export declare namespace juggler {
```

Comment:
AFAIK, namespaces are not needed when using Node.js `require` system, you can export types directly. 

```js
// loopback-datasource-juggler.ts
export type PromiseOrVoid<T> = Promise<T> | void;

// usage
import * as juggler from './loopback-datasource-juggler';
import {PromiseOrVoid} from './loopback-datasource-juggler';
```

Additionally, can we move this file into loopback-datasource-juggler module, to keep it in sync with the implementation? I am fine to do that later, after this initial pull request has landed.

Thoughts?